hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/spark/sql/hudi/command/procedures/ShowColumnStatsOverlapProcedure.scala

```diff
@@ -0,0 +1,355 @@
+                          fileSlicesSizeByPartition: Map[String, Int],
+                          rows: util.ArrayList[Row]): Unit = {
+    groupedPoints.map { case ((partitionPath, columnName), points) =>
+      val sortedPoints = points.sorted
+      var maxCount, currentCount = 0
+      val valueToCountMap: mutable.ListMap[String, Int] = mutable.ListMap.empty[String, Int]
+
+      sortedPoints.foreach { point =>
+        if (point.pType == "min") {
+          currentCount += 1
+          maxCount = Math.max(maxCount, currentCount)
+          valueToCountMap(point.value) = currentCount
+        } else {
+          if (!valueToCountMap.contains(point.value)) {
+            valueToCountMap(point.value) = currentCount
+          }
+          currentCount -= 1
+        }
+      }
+
+      val averageCount =
+        if (valueToCountMap.nonEmpty) valueToCountMap.values.sum.toDouble / valueToCountMap.size
+        else 0
+      val sortedCounts = valueToCountMap.values.toList.sorted
+
+      rows.add(Row(
+        partitionPath,
+        columnName,
+        averageCount,
+        maxCount,
+        fileSlicesSizeByPartition.get(partitionPath),
+        calculatePercentile(sortedCounts, 50),
+        calculatePercentile(sortedCounts, 75),
+        calculatePercentile(sortedCounts, 95),
+        calculatePercentile(sortedCounts, 99),
+        sortedCounts.size
+      ))
+    }
+  }
+
+  def calculatePercentile(values: List[Int], percentile: Double): Int = {
+    val index = (percentile / 100.0 * (values.size - 1)).toInt
+    values(index)
+  }
+
+  def buildFileSystemView(table: Option[Any]): HoodieTableFileSystemView = {
+    val basePath = getBasePath(table)
+    val metaClient = HoodieTableMetaClient.builder.setConf(jsc.hadoopConfiguration()).setBasePath(basePath).build
+    val fs = metaClient.getFs
+    val globPath = s"$basePath/*/*/*"
+    val statuses = FSUtils.getGlobStatusExcludingMetaFolder(fs, new Path(globPath))
+
+    val timeline = metaClient.getActiveTimeline.getCommitsTimeline.filterCompletedInstants()
+
+    val maxInstant = metaClient.createNewInstantTime()
+    val instants = timeline.getInstants.iterator().asScala.filter(_.getTimestamp < maxInstant)
+
+    val details = new Function[HoodieInstant, org.apache.hudi.common.util.Option[Array[Byte]]]
+      with java.io.Serializable {
+      override def apply(instant: HoodieInstant): HOption[Array[Byte]] = {
+        metaClient.getActiveTimeline.getInstantDetails(instant)
+      }
+    }
+
+    val filteredTimeline = new HoodieDefaultTimeline(
+      new java.util.ArrayList[HoodieInstant](JavaConversions.asJavaCollection(instants.toList)).stream(), details)
+
+    new HoodieTableFileSystemView(metaClient, filteredTimeline, statuses.toArray(new Array[FileStatus](statuses.size)))
+  }
+
+  def getColumnStatsValue(stats_value: Any): Option[String] = {
+    if (stats_value == null) {
+      logInfo("invalid value " + stats_value)
+      None
```

Comment:
Guess we can abstract this out as a communial utility.

Reply:
Done, this has been abstracted into HoodieTableMetadataUtil.